### ISSUE TITLE
feature request : use json natively

### ISSUE THREAD

**Issue by jmls** [chiefy/vaulted#46]
Wherever HCL is used, would it be possible to use json instead and internally convert to hcl before sending to vault ? It would make life _so_ much easier

so instead of having to use (taken from the default policy)

``` javascript
return vault.createPolicy({
    id: 'foo',
    body: {
      rules:"\npath \"auth/token/lookup-self\" {\n\tpolicy = \"read\"\n}\n\npath \"auth/token/renew-self\" {\n\tpolicy = \"write\"\n}\n\npath \"auth/token/revoke-self\" {\n\tpolicy = \"write\"\n}\n"}
    }
})
```

we could do

``` javascript
return vault.createPolicy({
    id: 'foo',
    body: {
      rules: [ { path: "auth/token/lookup-self", policy: "read" },
               { path: "auth/token/renew-self", policy: "write" },
               { path: "auth/token/revoke-self", policy: "write" }
             ]
    }

  })
```

**Comment by kenjones-cisco** [comment]
As noted in the Vault documentation (https://vaultproject.io/docs/concepts/policies.html) the Policies can be in either HCL or JSON. The documentation is not as clear through out as Vault itself is still maturing. In the Consul secret backend, it talks more about it and then refers to this url in the Consul documentation (https://consul.io/docs/internals/acl.html), towards the bottom of that page it shows what the HCL and JSON format of a policy look like.

HCL is a separate project on github that each of the Hashicorp products leverage for configuration.

I will try test this out in Vaulted today to make sure that is working as expected.

**Comment by jmls** [comment]
not quite - the link you pointed to is for acl _files_. The http api and vaulted client require a _string_ to be passed (either hcl or stringified json).

What I was asking for is that vaulted allows you to specify json, but it is stringified internally. Just makes my code cleaner ;)

**Comment by chiefy** [comment]
My :moneybag: is on supporting either a HCL string, or a JS object that we can just `JSON.stringify` as @jmls says.

**Comment by kenjones-cisco** [comment]
Resolved with commit 4c15330722c738d73bafef2d6250075a43040adc

Example input:

``` javascript
      var policy = {
        "path": {
          "secret/*": {
            "policy": "write"
          }
        }
      };
```
